### ISSUE TITLE
New PT2 Dashboard Tracker

### ISSUE THREAD

**Issue by desertfire** [pytorch/pytorch#96953]
We are building a new PT2 dashboard at https://hud.pytorch.org/benchmark/compilers

Based on the initial feedbacks, we have the following tasks to improve the dashboard,
- [x] Add UI for comparing two runs, P0 feature for oncall (@huydhn )
- [x] Small UI improvements, e.g. color coding for warning, compiler name changes, etc. (P0)(@huydhn )
- [x] Better UI for finding actual runner logs (P0)(@huydhn )
- [x] Simplify the nightly tests based on Ed's [proposal](https://docs.google.com/document/d/1jihUROLFah_Umx4962xRsp7xexqBAG268MVN66HRH-I/) to speedup perf collection (P0)(@desertfire )
- [x] Support comparing a PR run's result with a dashboard baseline and show it on the dashboard (P1)(@huydhn )
- [ ] Support auto-bisecting performance regression (P1)
- [ ] Send automatic alert to oncall when a regression happens (P1)  (@huydhn )
- [ ] Support commenting on dashboard for oncall to take note (P2)

Here are some further dashboard extensions to be done as the next step,
- [x] Add dynamic-shape tracking (@ezyang )
- [x] Add max-autotune tracking (@desertfire )
- [ ] Add 14k OSS model tracking (@yanboliang )

A note to keep track of some further feedback for the dashboard:

* https://github.com/pytorch/test-infra/issues/4002
* https://github.com/pytorch/test-infra/issues/4031

**Comment by huydhn** [comment]
During the demo, @l-kirsch mentioned about two weekly metrics used in addition to accuracy (IIRC)?  Do you know what they are @desertfire?  I didn't catch their names.

**Comment by desertfire** [comment]
It is https://github.com/pytorch/pytorch/issues/93667. @yanboliang collects those data weekly.

**Comment by Lokiiiiii** [comment]
I can't tell if the results are for training or inference. Also precision is missing.

**Comment by huydhn** [comment]
> I can't tell if the results are for training or inference. Also precision is missing.

Oh, this is training mode with amp precision.  `--inference` mode is not running in the benchmark at the moment I think (please correct me if that is not the case).  Is there a need to show both `amp` and `float32`? or just is having the former is ok like the existing dashboard https://github.com/pytorch/pytorch/issues/93794?

**Comment by desertfire** [comment]
>  Is there a need to show both `amp` and `float32`? or just is having the former is ok like the existing dashboard #93794?

There is actually another existing one for `float32`, https://github.com/pytorch/pytorch/issues/93518

**Comment by gchanan** [comment]
Can I link a particular run of the dashboard?  For example, if I have a conference or presentation I would like to "freeze" a certain run to report so I don't risk thrashing on bugs and such.

**Comment by desertfire** [comment]
> Can I link a particular run of the dashboard? For example, if I have a conference or presentation I would like to "freeze" a certain run to report so I don't risk thrashing on bugs and such.

Right now, you can get that by specifying the same commit id for both comparing branches, so the link would be something like `https://hud.pytorch.org/benchmark/torchbench/inductor_no_cudagraphs?mode=training&dtype=amp&lBranch=master&lCommit=c00fd71a95e0c52a14921e8d906d21d60dc9667c&rBranch=master&rCommit=c00fd71a95e0c52a14921e8d906d21d60dc9667c`. Do you mean you need one for overrall stats (like a snapshot of the front page)?

cc: @huydhn

**Comment by huydhn** [comment]
The example link that @desertfire shares is the correct way to get the snapshot of the the run.  I could add the same feature to the overall stats page, so the same could be done there

**Comment by ezyang** [comment]
If my browser window is too narrow, the Inductor link is not visible

<img width="1373" alt="image" src="https://user-images.githubusercontent.com/13564/231261130-9d9df1ef-b99b-4344-b729-571e6ebb676d.png">

**Comment by ezyang** [comment]
> The example link that @desertfire shares is the correct way to get the snapshot of the the run. I could add the same feature to the overall stats page, so the same could be done there

I would suggest adding a "share URL" link to permalink to a particular time.

**Comment by huydhn** [comment]
@gchanan @ezyang Just FYI, you can now take a snapshot of the dashboard and share the link that opens the exact same view

![Screenshot 2023-04-12 at 23 55 38](https://user-images.githubusercontent.com/475357/232174741-063645f5-b16a-498f-b08a-e3dd2b88f4a0.png)

**Comment by ezyang** [comment]
How exactly am I supposed to get the link, nothing seems to happen

**Comment by huydhn** [comment]
> How exactly am I supposed to get the link, nothing seems to happen

Oh, once you click on that, the snapshot link will be copied to the clipboard.  It can then be pasted/shared like a normal URL.
